crates/bevy_ecs/src/scheduling/executor/multi_threaded.rs

```diff
@@ -212,6 +200,20 @@ impl SystemExecutor for MultiThreadedExecutor {
                 scope.spawn(executor);
             },
         );
+
+        // SAFETY: all systems have completed
+        let world = unsafe { &mut *world.get() };
+        // this is running on the executor thread when it should run on the scope's thread
+        apply_system_buffers(&mut self.unapplied_systems, systems, world);
+        self.unapplied_systems.clear();
```

Comment:
Since you've fixed it, probably reword this as an invariant.

Reply:
Agreed on this comment.

Reply:
```suggestion
        // SAFETY: all systems have completed, and so no outstanding accesses remain
        let world = unsafe { &mut *world.get() };
        // Commands should be applied while on the scope's thread, not the executor's thread
        apply_system_buffers(&mut self.unapplied_systems, systems, world);
        self.unapplied_systems.clear();
```